js/components/CategoriesItem.jsx

```diff
@@ -7,11 +7,50 @@ var ScoreBar = require('./ScoreBar.jsx');
 var Link = Router.Link;
 
 var CategoriesItem = React.createClass({
+  getInitialState: function() {
+    return { investors: [] };
+  },
+
+  componentDidMount: function() {
+    this.getInvestors(this.props);
+  },
+
+  // Check to see if investors need to be refreshed
+  componentWillReceiveProps: function(props) {
+    if (props.userId !== this.props.userId) {
```

Comment:
Props are changed everytime you hover over the CategoriesItem. This caused a lot of unnecessary ajax calls, and the invariant violation (I think). Checking if the profile page has actual switched via the userId lets us avoid many many ajax calls